components/brave_ads/core/internal/common/resources/flat/text_classification_neural_model.fbs

```diff
@@ -0,0 +1,26 @@
+include "brave/components/brave_ads/core/internal/common/resources/flat/text_classification_transformation.fbs";
+
+namespace brave_ads.text_classification.flat;
+
+table WeightsRow {
+  row: [float32] (required);
+}
+
+table NeuralMatrixData {
+  weights_rows: [WeightsRow] (required);
+}
+
+table Classifier {
+  classes: [string] (required);
+  classifier_type: string (required);
+  neural_matricies_data: [NeuralMatrixData] (required);
+  neural_post_matrix_functions: [string] (required);
+}
+
+table NeuralModel {
+  classifier: Classifier (required);
+  locale: string (required);
```

Comment:
@ptjames @aseren do we validate that the users current language matches during classification?

Reply:
If a user changes system language, is the new language component resource not pulled (if not already available locally)? If not, that seems like it should be general behavior

Reply:
Yes they are but only if we have a supported language, but we should check that the resource code resets the resource for non supported locales so that we do not classify in those scenarios.

Reply:
added language code checking